api/hypershift/v1alpha1/nodepool_types.go

```diff
@@ -978,3 +982,23 @@ type Taint struct {
 	// Valid effects are NoSchedule, PreferNoSchedule and NoExecute.
 	Effect corev1.TaintEffect `json:"effect"`
 }
+
+// Diagnostics specifies the diagnostics settings for a virtual machine.
+// +kubebuilder:validation:XValidation:rule="self.storageAccountType == 'UserManaged' ? has(self.storageAccountURI) : true", message="storageAccountURI is required when storageAccountType is UserManaged"
+type Diagnostics struct {
+	// StorageAccountType determines if the storage account for storing the diagnostics data
+	// should be disabled (Disabled), provisioned by Azure (Managed) or by the user (UserManaged).
+	// +kubebuilder:validation:Enum=Managed;UserManaged;Disabled
+	// +kubebuilder:default:=Disabled
+	StorageAccountType string `json:"storageAccountType,omitempty"`
+	// StorageAccountURI is the URI of the user-managed storage account.
+	// The URI typically will be `https://<mystorageaccountname>.blob.core.windows.net/`
+	// but may differ if you are using Azure DNS zone endpoints.
+	// You can find the correct endpoint by looking for the Blob Primary Endpoint in the
+	// endpoints tab in the Azure console or with the CLI by issuing
+	// `az storage account list --query='[].{name: name, "resource group": resourceGroup, "blob endpoint": primaryEndpoints.blob}'`.
+	// +kubebuilder:validation:Pattern=`^https://`
```

Comment:
I am pretty sure kubebuilder has a uri format - is that appropriate to use here?

Reply:
Thinking of this maybe? https://kubernetes.io/docs/reference/using-api/cel/#kubernetes-url-library

Reply:
Ah yes, sorry - got my wires crossed

Reply:
This would consider absolute paths valid, not sure we want to allow that.

Reply:
Do you have an example of what you mean? I read the doc on their [absolute function](https://pkg.go.dev/net/url#URL.IsAbs) and it looks like it requires a scheme?

Reply:
So the function you linked only checks the scheme by the looks of it, this will allow the user to pass in an empty host and path. Not sure how this would work with a string as well. 

I think the validation on it atm is sufficient.

Reply:
The `isUrl()` validator uses `url.Parse()` from the stdlib - can you please be clear on what you're worried about here? Your regex-based validation would allow me to pass `"https://>@>#>$?%?#?%?#??#?"